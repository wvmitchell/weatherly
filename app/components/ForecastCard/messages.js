/*
 * ForecastCard Messages
 *
 * This contains all the text for the ForecastCard component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.ForecastCard.header',
    defaultMessage: 'This is the ForecastCard component !',
  },
});
