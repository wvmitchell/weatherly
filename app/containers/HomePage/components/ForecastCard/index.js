/**
*
* ForecastCard
*
*/

import React from 'react';

import Card, {
  CardHeader,
  CardContent,
} from 'material-ui/Card';
import {
  Avatar,
  Typography,
} from 'material-ui';

class ForecastCard extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <Card className="upcoming-weather">
        <CardHeader
          avatar={
            <Avatar
              alt={this.props.forecast.txt.forecastday[this.props.day * 2].icon}
              src={this.props.forecast.txt.forecastday[this.props.day * 2].icon_url}
            />
          }
          title={this.props.forecast.simple.forecastday[this.props.day].date.weekday}
          subheader={this.props.getTempExtremesString(this.props.forecast, this.props.day)}
        />
        <CardContent>
          <Typography type="headline" component="h3">
            Conditions
          </Typography>
          <Typography type="body1">
            {this.props.forecast.txt.forecastday[this.props.day * 2].fcttext}
          </Typography>
        </CardContent>
      </Card>
    );
  }
}

ForecastCard.propTypes = {
  forecast: React.PropTypes.object,
  getTempExtremesString: React.PropTypes.func,
  day: React.PropTypes.number,
};

export default ForecastCard;
