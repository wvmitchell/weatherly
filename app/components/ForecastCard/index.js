/**
*
* ForecastCard
*
*/

import React from 'react';

import Card, {
  CardHeader,
  CardMedia,
  CardContent,
} from 'material-ui/Card';
import messages from './messages';

class ForecastCard extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <Card className="todays-weather">
      </Card>
    );
  }
}

ForecastCard.propTypes = {
  forecast: React.PropTypes.object,
  day: React.PropTypes.number,
  night: React.PropTypes.number,
};

export default ForecastCard;
