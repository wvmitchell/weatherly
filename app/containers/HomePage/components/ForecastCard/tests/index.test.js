import React from 'react';
import { shallow } from 'enzyme';

import ForecastCard from '../index';

describe('<ForecastCard />', () => {
  it('should match the snapshot', () => {
    const mockForecast = {
      txt: {
        forecastday: [
          { icon: 'some-name', icon_url: 'some-path' },
        ],
      },
      simple: {
        forecastday: [
          {
            date: {
              weekday: 'sunday',
            },
          },
        ],
      },
    };
    const renderedComponent = shallow(
      <ForecastCard
        forecast={mockForecast}
        day={0}
        getTempExtremesString={jest.fn()}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });
});
