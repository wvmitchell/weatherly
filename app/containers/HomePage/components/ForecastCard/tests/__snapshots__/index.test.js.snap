exports[`<ForecastCard /> should match the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Card
    className="upcoming-weather"
    raised={false}>
    <withStyles(CardHeader)
        avatar={
            <withStyles(Avatar)
                alt="some-name"
                src="some-path" />
        }
        subheader={undefined}
        title="sunday" />
    <withStyles(CardContent)>
        <withStyles(Typography)
            component="h3"
            type="headline">
            Conditions
        </withStyles(Typography)>
        <withStyles(Typography)
            type="body1" />
    </withStyles(CardContent)>
</Card>,
  "nodes": Array [
    <Card
      className="upcoming-weather"
      raised={false}>
      <withStyles(CardHeader)
            avatar={
                  <withStyles(Avatar)
                        alt="some-name"
                        src="some-path" />
            }
            subheader={undefined}
            title="sunday" />
      <withStyles(CardContent)>
            <withStyles(Typography)
                  component="h3"
                  type="headline">
                  Conditions
            </withStyles(Typography)>
            <withStyles(Typography)
                  type="body1" />
      </withStyles(CardContent)>
</Card>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <ForecastCard
        day={0}
        forecast={
                Object {
                        "simple": Object {
                          "forecastday": Array [
                            Object {
                              "date": Object {
                                "weekday": "sunday",
                              },
                            },
                          ],
                        },
                        "txt": Object {
                          "forecastday": Array [
                            Object {
                              "icon": "some-name",
                              "icon_url": "some-path",
                            },
                          ],
                        },
                      }
        }
        getTempExtremesString={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ForecastCard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "day": 0,
          "forecast": Object {
            "simple": Object {
              "forecastday": Array [
                Object {
                  "date": Object {
                    "weekday": "sunday",
                  },
                },
              ],
            },
            "txt": Object {
              "forecastday": Array [
                Object {
                  "icon": "some-name",
                  "icon_url": "some-path",
                },
              ],
            },
          },
          "getTempExtremesString": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Card
          className="upcoming-weather"
          raised={false}>
          <withStyles(CardHeader)
                    avatar={
                              <withStyles(Avatar)
                                        alt="some-name"
                                        src="some-path" />
                    }
                    subheader={undefined}
                    title="sunday" />
          <withStyles(CardContent)>
                    <withStyles(Typography)
                              component="h3"
                              type="headline">
                              Conditions
                    </withStyles(Typography)>
                    <withStyles(Typography)
                              type="body1" />
          </withStyles(CardContent)>
</Card>,
        "_debugID": 2,
        "_renderedOutput": <Card
          className="upcoming-weather"
          raised={false}>
          <withStyles(CardHeader)
                    avatar={
                              <withStyles(Avatar)
                                        alt="some-name"
                                        src="some-path" />
                    }
                    subheader={undefined}
                    title="sunday" />
          <withStyles(CardContent)>
                    <withStyles(Typography)
                              component="h3"
                              type="headline">
                              Conditions
                    </withStyles(Typography)>
                    <withStyles(Typography)
                              type="body1" />
          </withStyles(CardContent)>
</Card>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ForecastCard
    day={0}
    forecast={
        Object {
            "simple": Object {
              "forecastday": Array [
                Object {
                  "date": Object {
                    "weekday": "sunday",
                  },
                },
              ],
            },
            "txt": Object {
              "forecastday": Array [
                Object {
                  "icon": "some-name",
                  "icon_url": "some-path",
                },
              ],
            },
          }
    }
    getTempExtremesString={[Function]} />,
}
`;
